### ISSUE TITLE
Show/hide chords and lyrics in score view

### ISSUE THREAD

**Issue by hugbug** [hugbug/conpianist#65]
I'm not sure if this is settable in Lomse.

**Comment by cecilios** [comment]
- For lirycs, currently no settings. But should not be complex to implement.
- For chords, what do you mean? chord symbols such as Cº? Lomse does not yet implement chord symbols.

**Comment by hugbug** [comment]
Screenshot from Smart Pianist:

![Bildschirmfoto 2020-04-22 um 19 57 36](https://user-images.githubusercontent.com/3368402/80016671-b81b8a80-84d3-11ea-87cc-9629a25ec0fc.png)

Can chords be embedded into MusicXML-files? And also important is that most of the MusicXML-files we use are generated from MIDI-files using MuseScore. I don't think chords are present in MIDI-files directly. Therefore MuseScore would need to detect them just like it builds score from MIDI-events.

The question is if MuseScore can build chords when importing MIDI-files. Because if it doesn't then we probably don't need the feature in Lomse. I think it's unlikely that users will add chords manually to the score (seems to be a task requiring some music knowledge).

**Comment by cecilios** [comment]
It is necessary to open the MusicXML file in a text editor to see how the chord symbol is exported by MuseScore, that is, to see which xml tag has been used. If it uses the _&lt;harmony&gt;_ symbol, then no problem. Lomse does not yet import them. But, most probably, it is exported as an attached text. In that case, to hide chord symbols it would be necessary to hide attached texts and that can affect also to ther notations.

**Comment by djangobaer** [comment]
There are different ways to implement chords in a midifile:
- you can insert a chordname. It is like a text. In Muse it is shown as a chord.
- you can insert a chord as Metacoding.
- you can insert a chord as Sysex, but I could not find that chordtype in Muse.
- you can insert a chord by notes, with some helping tools. (Cubase)
See pics.

Harald
![chord_in_MIDI_File_MixMaster](https://user-images.githubusercontent.com/62076869/80079271-5a795380-8550-11ea-922f-cd340f28254c.jpg)
![chord_in_MIDI_File_MixMaster_insert](https://user-images.githubusercontent.com/62076869/80079275-5baa8080-8550-11ea-909f-2ac1e642873a.jpg)
![chord_in_Musescore](https://user-images.githubusercontent.com/62076869/80079277-5baa8080-8550-11ea-99a9-06cff31cbfd8.jpg)
![chord_META_in_MIDI_File_MixMaster](https://user-images.githubusercontent.com/62076869/80079279-5c431700-8550-11ea-95c8-d976c10470c4.jpg)
![chord_META_in_Musescore](https://user-images.githubusercontent.com/62076869/80079281-5c431700-8550-11ea-8814-f743061750be.jpg)

**Comment by cecilios** [comment]
But here the important thing is to find how the chord symbol/name is encoded in the MusicXML file created by MuseScore when importing a MIDI file. MusicXML tag _&lt;chord/&gt;_ is not for chord symbol, it just marks the note to be grouped with other notes to form a chord.

**Comment by djangobaer** [comment]
You are right, there is no convertion of chordnames from midi to xml ba Musescore.
You can set chordnames in Muse and they will be shown in muse(harmony), but in the exportfile to midi there is nothing to find. In ConPianist (Lomse-score) you can see nothing from a chord.
In Smart Pianist there are the chords shown. So they take it out from midi. It will be helpful to see the chords made in muse(harmony) in Lomse, if possible.

**Comment by cecilios** [comment]
@djangobaer  Sorry, I do not understand:
> there is no convertion of chordnames from midi to xml ba Musescore

Does MuseScore export chord names to MusicXML or not?

Lomse only deals with MusicXML, but neither with MIDI nor with MuseScore.

**Comment by andrei-sage** [comment]
What is "muse(harmony)", is it a software to create MusicXML-files? If it can create MusicXML-files with chords can you please attach an example file?

**Comment by djangobaer** [comment]
MuseScore exports to MusicXML all chords, you have done in MuseScore as "harmony".

MusicXML-File: (as txt please rename for use)

![image](https://user-images.githubusercontent.com/62076869/80095877-f4e59100-8568-11ea-833f-e743be7ac783.png)

[Test_chords_D.txt](https://github.com/hugbug/conpianist/files/4522255/Test_chords_D.txt)

**Comment by cecilios** [comment]
Ok, when editing scores with MuseScore and saving the result as MusicXML chord names are saved as _harmony_ elements. Currently Lomse does not imports the _harmony_ elements so they will not be displayed.

Anyway, it should be clarified if MuseScore export the chord names when importing MIDI-files. Otherwise it will be irrelevant if lomse imports or not chord names.

@hugbug  When all the needs to solve this issue are confirmed, could you please file separated issues in the lomse repository (https://github.com/lenmus/lomse/issues) for each need? e.g. one for option to hide/show lyrics and chord names. And another for importing _harmony_ tags. Or wathever is needed. Thank you!
